secboot/secboot.go

```diff
@@ -251,3 +253,50 @@ func MarkSuccessful() error {
 
 	return nil
 }
+
+// AuthMode corresponds to an authentication mechanism.
+type AuthMode string
+
+const (
+	AuthModePassphrase AuthMode = "passphrase"
+	// TODO: Add PIN option when secboot support lands.
+)
+
+// VolumesAuthOptions contains options for the volumes authentication
+// mechanism (e.g. passphrase authentication).
+type VolumesAuthOptions struct {
+	Mode       AuthMode `json:"mode,omitempty"`
+	Passphrase string   `json:"passphrase,omitempty"`
+	// TODO: Add PIN option when secboot support lands.
+	KDFType string        `json:"kdf-type,omitempty"`
+	KDFTime time.Duration `json:"kdf-time,omitempty"`
+}
+
+// Validates authentication options.
+func (o *VolumesAuthOptions) Validate() error {
+	if o == nil {
+		return nil
+	}
+
+	switch o.Mode {
+	case AuthModePassphrase:
+		// TODO: Add entropy/quality checks on passphrase.
+		if len(o.Passphrase) == 0 {
+			return fmt.Errorf("passphrase cannot be empty")
+		}
+	default:
```

Comment:
and either explicitly reject pin here, or fall into the default handling

Reply:
I also added a bit of validation according to the spec kdf-type is not supported for PINs.